include/fmt/format.h

```diff
@@ -38,6 +38,7 @@
 #include <limits>        // std::numeric_limits
 #include <memory>        // std::uninitialized_copy
 #include <stdexcept>     // std::runtime_error
+#include <streambuf>     // std::basic_streambuf
```

Comment:
Adding a dependency on `streambuf` is undesirable esp. since `formatbuf` is not even used in `format.h`. Let's drop it by making `formatbuf` parameterized on `streambuf`.

Reply:
Done